docs/core/docker/publish-as-container.md

```diff
@@ -0,0 +1,349 @@
+
+```dotnetcli
+dotnet new worker -o Worker -n DotNet.ContainerImage
+```
+
+Your folder tree will look like the following:
+
+```Directory
+📁 sample-directory
+    └──📂 Worker
+        ├──appsettings.Development.json
+        ├──appsettings.json
+        ├──DotNet.ContainerImage.csproj
+        ├──Program.cs
+        ├──Worker.cs
+        └──📂 obj
+            ├── DotNet.ContainerImage.csproj.nuget.dgspec.json
+            ├── DotNet.ContainerImage.csproj.nuget.g.props
+            ├── DotNet.ContainerImage.csproj.nuget.g.targets
+            ├── project.assets.json
+            └── project.nuget.cache
+```
+
+The `dotnet new` command creates a new folder named *Worker* and generates a worker service that when ran will log a message every second. Change directories and navigate into the *Worker* folder, from your terminal session. Use the `dotnet run` command to start the app. The application will run, and print `Hello World!` below the command:
+
+```dotnetcli
+dotnet run
+Building...
+info: DotNet.ContainerImage.Worker[0]
+      Worker running at: 10/18/2022 08:56:00 -05:00
+info: Microsoft.Hosting.Lifetime[0]
+      Application started. Press Ctrl+C to shut down.
+info: Microsoft.Hosting.Lifetime[0]
+      Hosting environment: Development
+info: Microsoft.Hosting.Lifetime[0]
+      Content root path: .\Worker
+info: DotNet.ContainerImage.Worker[0]
+      Worker running at: 10/18/2022 08:56:01 -05:00
+info: DotNet.ContainerImage.Worker[0]
+      Worker running at: 10/18/2022 08:56:02 -05:00
+info: DotNet.ContainerImage.Worker[0]
+      Worker running at: 10/18/2022 08:56:03 -05:00
+info: Microsoft.Hosting.Lifetime[0]
+      Application is shutting down...
+Attempting to cancel the build...
+```
+
+The worker template loops indefinitely. Use the cancel command <kbd>Ctrl+C</kbd> to stop it.
+
+## Add NuGet package
+
+The .NET 7 SDK will eventually be capable of publishing .NET apps as containers without the use of the [Microsoft.NET.Build.Containers NuGet package](https://libraries.io/nuget/Microsoft.NET.Build.Containers). Until that time, this package is required. To add the `Microsoft.NET.Build.Containers` NuGet package to the worker template, run the following [dotnet add package](../tools/dotnet-add-package.md) command:
+
+```dotnetcli
+dotnet add package Microsoft.NET.Build.Containers
+```
+
+## Set the container image name
+
+There are various configuration options available when publishing an app as a container. For more information, see [Configure container image](#configure-container-image).
+
+By default, the container image name is the `AssemblyName` of the project. If that name is invalid as a container image name, you can override it by specifying a `ContainerImageName` as shown in the following:
```

Comment:
We do some normalization of application names by default (logging an informational message about the changes), so I think most users shouldn't _have_ to make this change if they don't want to.

Reply:
When I tried this locally, I was running into this error. I added the section to the article that details how to configure the name, as out-of-the-box it didn't seem to normalize `DotNet.ContainerImage`. With this name, I was experiencing that error.